src/effects/VST3/VST3EffectsModule.cpp

```diff
@@ -0,0 +1,290 @@
+class VST3PluginTraverser final : public wxDirTraverser
+{
+   std::function<void(const wxString&)> mOnPluginFound;
+public:
+   VST3PluginTraverser(std::function<void(const wxString&)> onPluginFound)
+      : mOnPluginFound(std::move(onPluginFound))
+   {
+   }
+
+   wxDirTraverseResult OnFile(const wxString& filename) override
+   {
+      if(filename.Matches("*.vst3"))
+         mOnPluginFound(filename);
+      return wxDIR_CONTINUE;
+   }
+   wxDirTraverseResult OnDir(const wxString& dirname) override
+   {
+      if(dirname.Matches("*.vst3"))
+      {
+         mOnPluginFound(dirname);
+         return wxDIR_IGNORE;//do not look inside...
+      }
+      return wxDIR_CONTINUE;
+   }
+};
+
+std::shared_ptr<VST3::Hosting::Module> VST3EffectsModule::GetModule(const wxString& path)
+{
+   const auto it = mModules.find(path);
+   if(it != mModules.end())
+   {
+      if(auto lock = it->second.lock())
+         return lock;
+   }
+
+   std::string moduleCreateError;
+   //VST sdk provides platform-specific module loading routines as well,
+   //implementation is conditionally included (see CMakeLists.txt)
+   auto module = VST3::Hosting::Module::create(path.ToStdString(), moduleCreateError);
+   if(!module)
+      throw std::runtime_error(moduleCreateError.c_str());
+
+   mModules[path] = module;
+   return module;
+}
+
+PluginPath VST3EffectsModule::GetPath()
+{
+   return {};
+}
+
+ComponentInterfaceSymbol VST3EffectsModule::GetSymbol()
+{
+   return XO("VST3 Effects");
+}
+
+VendorSymbol VST3EffectsModule::GetVendor()
+{
+   return XO("The Audacity Team");
+}
+
+wxString VST3EffectsModule::GetVersion()
+{
+   return AUDACITY_VERSION_STRING;
+}
+
+TranslatableString VST3EffectsModule::GetDescription()
+{
+   return XO("Adds the ability to use VST3 effects in Audacity.");
+}
+
+bool VST3EffectsModule::Initialize()
+{
+   return true;
+}
+
+void VST3EffectsModule::Terminate()
+{
+}
+
+EffectFamilySymbol VST3EffectsModule::GetOptionalFamilySymbol()
+{
+   return VST3Effect::GetFamilySymbol();
+}
+
+const FileExtensions& VST3EffectsModule::GetFileExtensions()
+{
+   static const FileExtensions ext {
+      { _T("vst3") }
+   };
+   return ext;
+}
+
+FilePath VST3EffectsModule::InstallPath()
+{
+#ifdef VST3_DEFAULT_INSTALL_PATH
+   return FilePath { VST3_DEFAULT_INSTALL_PATH };
+#else
+   return {};
+#endif
+}
+
+bool VST3EffectsModule::AutoRegisterPlugins(PluginManagerInterface& pluginManager)
+{
+   return false;
+}
+
+PluginPaths VST3EffectsModule::FindPluginPaths(PluginManagerInterface& pluginManager)
+{
+   //Note: The host recursively scans these folders at startup in this order (User/Global/Application).
+   //https://developer.steinberg.help/display/VST/Plug-in+Locations
+
+   FilePaths pathList;
+#ifdef __WXMSW__
+   // Windows specific VST3 search paths
+   {
+      wxString programFilesPath;
+      if(wxGetEnv("programfiles", &programFilesPath))
+         pathList.push_back(programFilesPath + "\\Common Files\\VST3");
+   }
```

Comment:
I see you have removed a check in "programfiles(x86)" that was in an earlier version of the PR.  No mistake?

Reply:
Yes, there is no need to search x86 plugins in x64 bit Audacity as they will not be loaded anyway